### PR TITLE
ITNFA-24: Disabled fingerprint scan button, when processing

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-function m2sysEnroll() {
+function m2sysEnroll(button) {
+    toggleFingerprintButtonDisplay(button);
     jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/enroll.action')
     .success(function(data) {
+        toggleFingerprintButtonDisplay(button);
```
Comment (areklalo) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: What about the failed messages. I think we should also toggle the button when  we get .error() or .fail(). Perhaps you can use .always() or .done()

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 2]
```diff
@@ -18,13 +20,15 @@ function m2sysGetStatus() {
...
-function m2sysUpdate(idValue) {
+function m2sysUpdate(idValue, button) {
+    toggleFingerprintButtonDisplay(button);
     jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/update.action',
         {
             id: idValue
         }
     )
     .success(function(data) {
+        toggleFingerprintButtonDisplay(button);
```
Comment (areklalo) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: Like above

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js (hunk 1 above)
Comment (oskarhinc) [reply omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: You're right, I'll handle this case too.

**Review by areklalo — APPROVED** [review]
LGTM

File: omod/src/main/webapp/fragments/field/m2sysEnrollOrUpdate.gsp [hunk 3]
```diff
@@ -34,9 +36,9 @@
 		
 		jq(fingerprintButton).click(function() {
 			if (enrollOrUpdate == "update") {
-          		m2sysUpdate(biometricID);
+          		m2sysUpdate(biometricID, this);
         	} else {
-				m2sysEnroll();
+				m2sysEnroll(this);
```
Comment (dserkowski) [inline omod/src/main/webapp/fragments/field/m2sysEnrollOrUpdate.gsp]: Something wrong with indentation (in the whole file). Could you fix this?

File: omod/src/main/webapp/fragments/search/m2SysSearch.gsp [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
 <script>
 jq = jQuery;
 
 jq(function() {
-    jq('#fingerprint_search_button').click(function() {
+    var searchButton = jq('#fingerprint_search_button');
+    searchButton.click(function() {
+        toggleFingerprintButtonDisplay(searchButton);
         jq.getJSON('${ ui.actionLink("getPatients") }', {})
+        .always(function() {
```
Comment (dserkowski) [inline omod/src/main/webapp/fragments/search/m2SysSearch.gsp]: bad indentation

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 function m2sysEnroll(button) {
     toggleFingerprintButtonDisplay(button);
     jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/enroll.action')
-    .success(function(data) {
+    .always(function() {
```
Comment (dserkowski) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: bad indentation

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 6]
```diff
@@ -27,8 +29,10 @@ function m2sysUpdate(idValue, button) {
             id: idValue
         }
     )
-    .success(function(data) {
+    .always(function() {
```
Comment (dserkowski) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: bad indentation

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
-function m2sysEnroll() {
+function m2sysEnroll(button) {
+    toggleFingerprintButtonDisplay(button);
     jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/enroll.action')
+    .always(function() {
```
Comment (dserkowski) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: bad indentation

**Review by dserkowski — CHANGES_REQUESTED** [review]

**Comment by oskarhinc** [comment]
I've fixed the indentation in those files.

File: omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js [hunk 8]
```diff
@@ -1,73 +1,81 @@
...
-        }
-    );
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/delete.action',
+		{
+			id: idValue
+		}
+	);
+	//TODO add success method
 }
```
Comment (pawelcieszko) [inline omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: shouldn't a new line sign be added at the end?

https://stackoverflow.com/questions/4700312/empty-new-line-at-the-end-of-the-java-source-files/14637072#14637072

Comment (oskarhinc) [reply omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js]: Yes, I can do it

File: omod/src/main/webapp/fragments/field/m2sysEnrollOrUpdate.gsp [hunk 9]
```diff
@@ -1,61 +1,64 @@
...
-				        identifierValue: biometricID
-				    }
-				    , function(data) {
-				        emr.successMessage(data.message);
-				        enrollOrUpdate = "update";
+
+		jq(fingerprintSubjectIdField).change(function() {
+			biometricID = jq(fingerprintSubjectIdField).val();
+			if (biometricID) {
+			    emr.getFragmentActionWithCallback('coreapps', 'editPatientIdentifier', 'editPatientIdentifier',
```
Comment (dserkowski) [inline omod/src/main/webapp/fragments/field/m2sysEnrollOrUpdate.gsp]: Something is still not correct. Try this: https://stackoverflow.com/a/9872584

**Review by dserkowski — CHANGES_REQUESTED** [review]

**Review by dserkowski — APPROVED** [review]

**Review by pawelcieszko — APPROVED** [review]

**Pull request merged**